### PR TITLE
Allow for custom iTipBroker

### PR DESCRIPTION
**Pull request by SebastianKrupinski**
Uses a protected function to generate the iTipBroker so that is can be overriden and used to generate a custom broker class.

### CODE REVIEW

File: lib/CalDAV/Schedule/Plugin.php [hunk 1]
```diff
@@ -994,4 +994,14 @@ public function getPluginInfo()
             'link' => 'http://sabre.io/dav/scheduling/',
         ];
     }
+
+    /**
+     * Returns an instance of the iTip\Broker
+     * 
+     * @return Broker
```
Comment (ChristophWurst) [inline lib/CalDAV/Schedule/Plugin.php:1001]: Needs an important and please use a return type hint

Comment (ChristophWurst) [reply lib/CalDAV/Schedule/Plugin.php:1002]: Looks good now!

**Comment by ChristophWurst** [comment]
@phil-davis I hope this change makes sense. We add the protected factory method so that sub classes of the plugin can work with a sub-classed broker as well.

**Comment by SebastianKrupinski** [comment]
@DeepDiver1975 Done.

**Review by DeepDiver1975 — APPROVED** [review]

**Review by NameNoQuality — APPROVED** [review]

**Pull request merged**